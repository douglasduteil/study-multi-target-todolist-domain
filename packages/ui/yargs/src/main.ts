//

import yargs from "yargs";

export function main({ version }: { version: string }): yargs.Argv {
  return yargs
    .usage("Usage: $0 <command> [options]")
    .version(version)
    .demandCommand(
      1,
      "A command is required. Pass --help to see all available commands and options."
    )
    .recommendCommands()
    .strict()
    .alias("h", "help")
    .alias("v", "version");
}
