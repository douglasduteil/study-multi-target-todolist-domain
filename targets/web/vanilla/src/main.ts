//

import { Container } from "inversify";
import { IDENTIFIER } from "./app/identifiers";
import { AppModule } from "./app/module";

//

export async function bootstrap(): Promise<void> {
  const container = new Container();
  container.load(AppModule);

  // MUST auto bootstrap on "postConstruct"
  container.get(IDENTIFIER.ROOT_COMPONENT);
}
