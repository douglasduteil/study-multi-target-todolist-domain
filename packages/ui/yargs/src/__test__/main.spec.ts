//

import { main } from "../main";

//

test("should display main help", () => {
  const log = jest.fn();
  main({ version: "x.y.z" })
    .scriptName("test")
    .wrap(80)
    .parse([], log);
  const [[error, , output]] = log.mock.calls;
  expect(error.message).toMatchInlineSnapshot(
    `"A command is required. Pass --help to see all available commands and options."`
  );
  expect(output).toMatchSnapshot();
});

test("should display version", () => {
  const log = jest.fn();
  main({ version: "x.y.z" })
    .scriptName("test")
    .wrap(80)
    .parse(["--version"], log);
  const [[error, , output]] = log.mock.calls;
  expect(error).toBeUndefined();
  expect(output).toMatchInlineSnapshot(`"x.y.z"`);
});
