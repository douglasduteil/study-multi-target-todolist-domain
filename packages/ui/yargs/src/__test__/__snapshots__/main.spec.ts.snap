// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display main help 2`] = `
"Usage: test <command> [options]

Options:
  -h, --help     Show help                                             [boolean]
  -v, --version  Show version number                                   [boolean]

A command is required. Pass --help to see all available commands and options."
`;
